docs/source/tracking/tracking-api.rst

```diff
@@ -0,0 +1,281 @@
+
+Organizing Runs in Experiments
+------------------------------
+
+MLflow allows you to group runs under experiments, which can be useful for comparing runs intended
+to tackle a particular task. You can create experiments via multiple way - MLflow UI, the :ref:`cli` (``mlflow experiments``),
+or the :py:func:`mlflow.create_experiment` Python API. You can pass the experiment name for an individual run
+using the CLI (for example, ``mlflow run ... --experiment-name [name]``) or the ``MLFLOW_EXPERIMENT_NAME``
+environment variable. Alternatively, you can use the experiment ID instead, via the
+``--experiment-id`` CLI flag or the ``MLFLOW_EXPERIMENT_ID`` environment variable.
+
+.. code-block:: bash
+
+    # Set the experiment via environment variables
+    export MLFLOW_EXPERIMENT_NAME=fraud-detection
+
+    mlflow experiments create --experiment-name fraud-detection
+
+.. code-block:: python
+
+    # Launch a run. The experiment is inferred from the MLFLOW_EXPERIMENT_NAME environment
+    # variable, or from the --experiment-name parameter passed to the MLflow CLI (the latter
+    # taking precedence)
+    with mlflow.start_run():
+        mlflow.log_param("a", 1)
+        mlflow.log_metric("b", 2)
+
+.. _launching-multiple-runs:
+
+Launching Multiple Runs in One Program
+--------------------------------------
+Sometimes you want to launch multiple MLflow runs in the same program: for example, maybe you are
+performing a hyperparameter search locally or your experiments are just very fast to run. This is
+easy to do because the ``ActiveRun`` object returned by :py:func:`mlflow.start_run` is a Python
+`context manager <https://docs.python.org/2.5/whatsnew/pep-343.html>`_. You can "scope" each run to
+just one block of code as follows:
+
+.. code-block:: python
+
+   with mlflow.start_run():
+       mlflow.log_param("x", 1)
+       mlflow.log_metric("y", 2)
+       ...
+
+The run remains open throughout the ``with`` statement, and is automatically closed when the
+statement exits, even if it exits due to an exception.
+
+
+.. _add-tags-to-runs:
+
+Adding Tags to Runs
+-------------------
+
+You can annotate runs with arbitrary tags to organize them into groups. This allows you to easily filter and search
+Runs in Tracking UI by using :ref:`filter experssion <mlflow_tags>`.
+
+.. _system_tags:
+
+System Tags
+~~~~~~~~~~~
+
+Tag keys that start with ``mlflow.`` are reserved for internal use.
+The following tags are set automatically by MLflow, when appropriate:
+
++-------------------------------+----------------------------------------------------------------------------------------+
+| Key                           | Description                                                                            |
++===============================+========================================================================================+
+| ``mlflow.note.content``       | A descriptive note about this run. This reserved tag is not set automatically and can  |
```

Comment:
Can we call out in the text above the table that this is an Exception and is the best way to programmatically pre-seed the notes location? A LOT of new users ask about this.

Reply:
Ooh I see, yea definitely otherwise it's confusing to use reserved key. Also worth having subsection and FAQ imo.